src/seedu/addressbook/data/person/Name.java

```diff
@@ -61,4 +62,65 @@ public int hashCode() {
         return fullName.hashCode();
     }
 
+    /**
+     * Returns true if the other name is very similar to this name. 
+     * Two names are considered similar if: 
+     * <pre>
+     * 1. they are in different case
+     * 2. they are subset/superset of each other
+     * </pre>
+     */
+    public boolean isSimilar(Name other) {
+        if(!isOtherNameNull(other)) {
+            return isNameEqualsIgnoreCase(other) ||
+                 isNameSubsetOfOther(other) ||
```

Comment:
think how you can eliminate one the methods by using different objects and the same method!